dbms/src/Storages/KVStore/MultiRaft/ApplySnapshot.cpp

```diff
@@ -196,9 +222,9 @@ void KVStore::onSnapshot(
                             new_key_range.toDebugString(),
                             keyspace_id,
                             table_id);
-                        dm_storage->deleteRange(old_key_range, context.getSettingsRef());
+                        dm_storage->deleteRange(new_key_range, context.getSettingsRef());
                         // We must flush the deletion to the disk here, because we only flush new range when persisting this region later.
-                        dm_storage->flushCache(context, old_key_range, /*try_until_succeed*/ true);
+                        dm_storage->flushCache(context, new_key_range, /*try_until_succeed*/ true);
```

Comment:
Why here is changed to delete the new_key_range?

Reply:
Consider:
1. apply snapshot a of range [0..100)
2. apply snapshot b of range [50..100)
3. apply snapshot a' of range [0..50)

In 3, we could clean the old range [0..100), which covers the written data of snapshot b in stage 2

Reply:
If we only need to delete the existing data on the new_range, then the following code of `ingestFiles(..., /*clear_data_in_range=*/true, ...)` can do it. We don't need this branch.

Reply:
Fixed